File: algorithms/squares_of_sorted_array.py
def sortedSquares(A: List[int]) -> List[int]:
    index = len(A) - 1
    results = [0 for _ in A]
    negative, positive = 0, len(A) - 1
    while index >= 0:
        if abs(A[negative]) > A[positive]:
            results[index] = A[negative] ** 2
            negative += 1
        else:
            results[index] = A[positive] ** 2
            positive -= 1
        index -= 1
    return results

File: algorithms/squares_of_sorted_array.go
func sortedSquares(A []int) []int {
	results := make([]int, len(A))
	index := len(A) - 1
	negative, positive := 0, len(A) - 1

	for index >= 0 {
		if square(A[negative]) > square(A[positive]) {
			results[index] = square(A[negative])
			negative += 1
		} else {
			results[index] = square(A[positive])
			positive -= 1
		}

		index -= 1
	}

	return results
}

func square(n int) int {
	return n * n
}
